### DISCUSSION TITLE
Does next.js have SPA mode?

### DISCUSSION THREAD

**Discussion by dln88** [vercel/next.js#17863] [Help] [answered]
I have a project using React & NextJS. Currently, I have a custom _app.js that make the page become SSR. But now I want the page not being load on the server-side. Is there any way to do that?

**Comment by jamesmosier** [comment, marked as answer]
Next by default is a server rendered framework.

You can disable SSR individually by using [next/dynamic](https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr).

Also using [next/link](https://nextjs.org/docs/api-reference/next/link) to link between pages will make it navigate like a single page app.

**Reply by RezaRamezanpour** [reply]
Can we have something hybrid? for example SSR at first load(for SEO) and CSR for substitute requests like normal CRA?

**Reply by jamesmosier** [reply]
That's how Next works by default! It does SSR for first load and then if you use [`next/link`](https://nextjs.org/docs/api-reference/next/link) to navigate to other pages on your site it is client side rendered.

**Reply by RezaRamezanpour** [reply]
Aha. got it. I didn't notice that the link was client-side due to interruption in rendering while using `getServerSideProps`. Thanks a lot.

**Reply by berksafran** [reply]
@jamesmosier How to understand that CSR works well? (by using Chrome Browser or adding console.log codes)
